components/ModInputField.vue

```diff
@@ -10,42 +10,40 @@
             {{ invalidInputErrorMessage }}
         </p>
         <input
+            v-model="model"
             :type="type"
             :placeholder="placeholder"
             :required="required"
-            :value="inputValue"
             class="mb-5 px-3 py-3.5 w-96 h-12 bg-white rounded-lg border border-primary-text-muted
             text-primary-text text-sm font-normal font-sans placeholder-primar"
-            @blur="() => checkValidationOfInputField(inputValue)"
-            @input="handleInput"
+            @blur="() => checkValidationOfInputField(model as string)"
```

Comment:
improvement idea: you could make the validation smarter to run on change instead of blur.

Reply:
I could not remember exactly how we discussed this before but I updated it to on blur flip the Boolean and check the validation. Then used vue's built in watch in order to happen on change after the initial check with the model.